src/Symfony/Component/PropertyInfo/Util/PhpDocTypeHelper.php

```diff
@@ -74,8 +74,24 @@ public function getTypes(DocType $varType): array
     /**
      * Creates a {@see Type} from a PHPDoc type.
      */
-    private function createType(string $docType, bool $nullable): ?Type
+    private function createType(DocType $type, bool $nullable): ?Type
     {
+        $docType = (string) $type;
+
+        if ($type instanceof Collection) {
+            list($phpType, $class) = $this->getPhpTypeAndClass((string) $type->getFqsen());
+
+            $key = $this->getTypes($type->getKeyType());
+            $value = $this->getTypes($type->getValueType());
+
+            // More than 1 type returned means it is a Compound type, which is
```

Comment:
Do you think we can improve the type VO (it's a final class exactly for this reason) to support this?

Reply:
I guess it could be done without much shenanigans (adding a counpound internal type, and a property holding composing types).

But i'm afraid that would be a breaking change, as the component will start to return non-null types where it previously did.